File: sparkmall/spark-mock/src/main/scala/com/atguigu/sparkmall/mock/RandomDate.scala
package com.atguigu.sparkmall.mock

import java.util.Date

import scala.util.Random

object RandomDate {
  def apply(startDate:Date,endDate:Date,step:Int): RandomDate ={
    val randomDate = new RandomDate()
    val avgStepTime = (endDate.getTime- startDate.getTime)/step
    randomDate.maxTimeStep=avgStepTime*2
    randomDate.lastDateTime=startDate.getTime
    randomDate
  }
  def getRandomDateTime(startDate:Date,endDate:Date):Date = {
    val randomTime = startDate.getTime + new Random().nextInt((endDate.getTime - startDate.getTime).toInt)
    new Date(randomTime)
  }
}

class RandomDate{
  var lastDateTime =0L
  var maxTimeStep=0L

  def  getRandomDate()={
    val timeStep = new Random().nextInt(maxTimeStep.toInt)
    lastDateTime = lastDateTime+timeStep

    new Date( lastDateTime)
  }

}

File: gmail_DW/dw-mock/src/main/java/com/atguigu/dw/mock/util/RandomDate.java
package com.atguigu.dw.mock.util;

import java.util.Date;
import java.util.Random;

public class RandomDate {

    Long logDateTime =0L;//
    int maxTimeStep=0 ;

    public RandomDate() {
    }

    public RandomDate(Date startDate, Date endDate, int num){
        long avgStepTime = (endDate.getTime() - startDate.getTime()) / num;
        this.maxTimeStep = (int)avgStepTime*2;
        logDateTime = startDate.getTime();
    }

    public Date getRandomDate(){
        int timeStep = new Random().nextInt(maxTimeStep);
        logDateTime += timeStep;
        return new Date(logDateTime);
    }
}
